framework/sandstone.cpp

```diff
@@ -1038,6 +1039,43 @@ int num_packages() {
     return Topology::topology().packages.size();
 }
 
+int num_enabled_cpus() {
+    return sApp->enabled_cpus_count;
+}
+
+void reschedule() {
+    // Choose a random CPU to hop to
+    int cpu_idx = random32() % num_enabled_cpus();
+    int next_cpu = cpu_info[cpu_idx].cpu_number;
+
+    if (pin_to_logical_processor(LogicalProcessor(next_cpu)))
+        log_debug("Rescheduled thread %d (%ld) to CPU %d",thread_num, pthread_self(), next_cpu);
+    else
+        log_debug("Failed to reschedule %d (%ld) to CPU %d", thread_num, pthread_self(), next_cpu);
+
+    return;
+}
+
+void reschedule_queue(std::vector<int> *reschedule_cpu_q, std::atomic<int> *cpu_idx) {
```

Comment:
This cannot and need not to be C++. Also, it should be just 1 call -- reschedule() as I outlined in the other comment.